charmcraft/commands/build.py

```diff
@@ -79,6 +81,21 @@ def polite_exec(cmd):
     return retcode
 
 
+def relativise(src, dst):
+    """Build a relative path from src to dst."""
+    src_rel_parts = []
+    dst_rel_parts = []
+    for src_part, dst_part in itertools.zip_longest(src.parent.parts, dst.parts):
+        if src_part != dst_part:
+            if src_part is not None:
+                src_rel_parts.append('..')
+            if dst_part is not None:
+                dst_rel_parts.append(dst_part)
+
+    link = pathlib.Path(os.path.join(*(src_rel_parts + dst_rel_parts)))
```

Comment:
the `os.path.join` call seems spurious to me (all tests pass if I remove it)

also `*(a+b)` is the same as `*a, *b`, but that might be an aesthetic thing